### PR TITLE
Converted ZC_DELETE_RELATED_GUILD to struct

### PR DESCRIPTION
**Pull request by AoShinRO**
* **Addressed Issue(s)**: 

* **Server Mode**: 

* **Description of Pull Request**:

### CODE REVIEW

File: src/map/packets.hpp [hunk 1]
```diff
@@ -760,6 +760,13 @@ struct PACKET_ZC_OPEN_EDITDLGSTR {
 } __attribute__((packed));
 DEFINE_PACKET_HEADER(ZC_OPEN_EDITDLGSTR, 0x1d4)
 
+struct PACKET_ZC_DELETE_RELATED_GUILD {
+	int16 packetType;
+	int relationGuildId;
```
Comment (Daegaladh) [inline src/map/packets.hpp:765]:
```suggestion
	uint32 relationGuildId;
```

File: src/map/packets.hpp [hunk 2]
```diff
@@ -760,6 +760,13 @@ struct PACKET_ZC_OPEN_EDITDLGSTR {
...
+	int relationFlag;
```
Comment (Daegaladh) [inline src/map/packets.hpp:766]:
```suggestion
	uint32 relationFlag;
```

File: src/map/packets.hpp (hunk 1 above)
Comment (AoShinRO) [reply src/map/packets.hpp:765]: i followed emulator decl
image

Comment (Daegaladh) [reply src/map/packets.hpp:765]: should be the other way around, we always need to send the correct type for the packets

File: src/map/clif.cpp [hunk 3]
```diff
@@ -9198,25 +9198,23 @@ void clif_guild_allianceack(map_session_data *sd,int flag)
...
+/// 0184 <other guild id>.L <relation>.L (ZC_DELETE_RELATED_GUILD).
 /// relation:
 ///     0 = Ally
 ///     1 = Enemy
-void clif_guild_delalliance(map_session_data *sd,int guild_id,int flag)
+void clif_guild_delalliance(map_session_data& sd,int guild_id,int flag)
 {
-	nullpo_retv(sd);
+	if ( !session_isActive(sd.fd) )
+		return;
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:9209]:
```suggestion
```

File: src/map/clif.hpp [hunk 4]
```diff
@@ -1060,7 +1060,7 @@ void clif_guild_notice( map_session_data& sd );
 void clif_guild_message( const struct mmo_guild& g, uint32 account_id, const char* mes, size_t len );
 void clif_guild_reqalliance(map_session_data *sd,uint32 account_id,const char *name);
 void clif_guild_allianceack(map_session_data *sd,int flag);
-void clif_guild_delalliance(map_session_data *sd,int guild_id,int flag);
+void clif_guild_delalliance(map_session_data& sd,int guild_id,int flag);
```
Comment (Lemongrass3110) [inline src/map/clif.hpp:1063]:
```suggestion
void clif_guild_delalliance(map_session_data& sd,uint32 guild_id,uint32 flag);
```

File: src/map/clif.cpp [hunk 5]
```diff
@@ -9198,25 +9198,23 @@ void clif_guild_allianceack(map_session_data *sd,int flag)
...
 
-/// Notifies the client that a alliance or opposition has been removed (ZC_DELETE_RELATED_GUILD).
-/// 0184 <other guild id>.L <relation>.L
+/// Notifies the client that a alliance or opposition has been removed 
+/// 0184 <other guild id>.L <relation>.L (ZC_DELETE_RELATED_GUILD).
 /// relation:
 ///     0 = Ally
 ///     1 = Enemy
-void clif_guild_delalliance(map_session_data *sd,int guild_id,int flag)
+void clif_guild_delalliance(map_session_data& sd,int guild_id,int flag)
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:9206]:
```suggestion
void clif_guild_delalliance(map_session_data& sd,uint32 guild_id,uint32 flag)
```

File: src/map/clif.cpp [hunk 6]
```diff
@@ -9198,25 +9198,23 @@ void clif_guild_allianceack(map_session_data *sd,int flag)
...
+	if ( !session_isActive(sd.fd) )
+		return;
 
-	int fd = sd->fd;
+	PACKET_ZC_DELETE_RELATED_GUILD p{};
 
-	if ( !session_isActive(fd) )
-		return;
+	p.packetType = HEADER_ZC_DELETE_RELATED_GUILD;
+	p.allyID = static_cast<decltype(p.allyID)>(guild_id);
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:9214]:
```suggestion
	p.allyID = guild_id;
```

File: src/map/clif.cpp [hunk 7]
```diff
@@ -9198,25 +9198,23 @@ void clif_guild_allianceack(map_session_data *sd,int flag)
...
+	p.flag = static_cast<decltype(p.flag)>(flag);
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:9215]:
```suggestion
	p.flag = flag;
```

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/clif.cpp [hunk 8]
```diff
@@ -9195,25 +9195,21 @@ void clif_guild_allianceack(map_session_data *sd,int flag)
 }
 
 
-/// Notifies the client that a alliance or opposition has been removed (ZC_DELETE_RELATED_GUILD).
-/// 0184 <other guild id>.L <relation>.L
+/// Notifies the client that a alliance or opposition has been removed 
+/// 0184 <other guild id>.L <relation>.L (ZC_DELETE_RELATED_GUILD).
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:9199]:
```suggestion
/// Notifies the client that a alliance or opposition has been removed.
/// 0184 <other guild id>.L <relation>.L (ZC_DELETE_RELATED_GUILD)
```

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/packets.hpp [hunk 9]
```diff
@@ -760,12 +760,20 @@ struct PACKET_ZC_OPEN_EDITDLGSTR {
 } __attribute__((packed));
 DEFINE_PACKET_HEADER(ZC_OPEN_EDITDLGSTR, 0x1d4)
 
+
```
Comment (AoShinRO) [inline src/map/packets.hpp:763]:
```suggestion

```

**Pull request merged**